File: js/lastStoneWeight.js

const lastStoneWeight = (stones) => {
    while (stones.length > 1) {
        let s2 = Math.max(...stones);
        stones.splice(stones.indexOf(s2), 1);
        let s1 = Math.max(...stones);
        stones.splice(stones.indexOf(s1), 1);
        if (s1 == s2) {
            continue;
        } else {
            stones.push(s2 - s1);
        }
    }
    return stones.length > 0 ? stones[0] : 0;
}

const result = lastStoneWeight([2, 7, 4, 1, 8, 1]);

File: py/lastStoneWeight.py

def lastStoneWeight(stones):

    while len(stones) > 1:
        s2 = max(stones)
        stones.remove(s2)
        s1 = max(stones)
        stones.remove(s1)
        if s1 == s2:
            continue
        else:
            stones.append(s2 - s1)

    if len(stones) > 0:
        return stones[0]
    else:
        return 0


result = lastStoneWeight([2,7,4,1,8,1])